app/controllers/admin/spaces_controller.rb

```diff
@@ -40,7 +40,20 @@ def create_sub_space
 
   def delete_sub_space
     if params[:id].present?
-      if SubSpace.find(params[:id]).delete
+      subspace = SubSpace.find(params[:id])
+      SubSpaceBooking
+        .where(sub_space: subspace)
+        .each do |booking|
+          status =
+            SubSpaceBookingStatus.find(booking.sub_space_booking_status_id)
+          status.sub_space_booking_id = nil
```

Comment:
Could we use `.update()` for these instead? It would save a line each time and is also easier to read